Exercises/Philipp Schöneberg/exercises_02_solved.py

```
@@ -0,0 +1,75 @@
+# After the program stops, the function should return the value in the first location (address 0) 
+# of your working memory.
+
+# As an example, if the list of integers passed to your function is 
+# [1, 0, 0, 0, 99] the 1 in the first position indicates you should read the values
+# at position given by the second and third entries. Both of these indicate position 0, so you should read the value
+# at position 0 twice. That value is 1. Adding 1 and 1 gives you two. You then look at the value in the fourth
+# position, which is again 0, so you write the result to position 0. You then step forward by 4 steps, arriving at 99
+# and ending the program. The final memory looks like [2, 0, 0, 0, 99]. Your function should return 2.
+
+# Here's another testcase:
+# [1, 1, 1, 4, 99, 5, 6, 0, 99] should become [30, 1, 1, 4, 2, 5, 6, 0, 99]
+# Your function should return 30.
+
+
+# print out which value is returned by your function for the following list:
+commands = [1, 12, 2, 3, 1, 1, 2, 3, 1, 3, 4, 3, 1, 5, 0, 3, 2, 1, 9, 19, 1, 5, 19, 23, 1, 6, 23, 27, 1, 27, 10, 31, 1, 31, 5, 35, 2, 10, 35, 39, 1, 9, 39, 43, 1, 43, 5, 47, 1, 47, 6, 51, 2, 51, 6, 55, 1, 13, 55, 59, 2, 6, 59, 63, 1, 63, 5, 67, 2, 10, 67, 71, 1, 9, 71, 75, 1, 75, 13, 79, 1, 10, 79, 83, 2, 83, 13, 87, 1, 87, 6, 91, 1, 5, 91, 95, 2, 95, 9, 99, 1, 5, 99, 103, 1, 103, 6, 107, 2, 107, 13, 111, 1, 111, 10, 115, 2, 10, 115, 119, 1, 9, 119, 123, 1, 123, 9, 127, 1, 13, 127, 131, 2, 10, 131, 135, 1, 135, 5, 139, 1, 2, 139, 143, 1, 143, 5, 0, 99, 2, 0, 14, 0]
+
+def lst_manipulator (lst):
+    for i, elem in enumerate(lst[::4]):
+        match lst[i*4]:
+            case 1:
+                lst[lst[i*4+3]] = lst[lst[i*4+1]]+lst[lst[i*4+2]]
+            case 2:
+                lst[lst[i*4+3]] = lst[lst[i*4+1]]*lst[lst[i*4+2]]
+            case 99:
+                break
+            case _:
+                print("The given opcode is unvalid.")
```

Maybe raise an error like you did for the second task.

---

Good idea I just did. Thx!